libmultilabel/linear/linear.py

```diff
@@ -475,6 +472,43 @@ def train_cost_sensitive_micro(y: sparse.csr_matrix, x: sparse.csr_matrix, optio
     return {'weights': np.asmatrix(weights), '-B': bias, 'threshold': 0}
 
 
+def train_binary_and_multiclass(y: sparse.csr_matrix, x: sparse.csr_matrix, options: str):
+    """Trains a linear model for binary and multi-class data.
+
+    Args:
+        y (sparse.csr_matrix): A 0/1 matrix with dimensions number of instances * number of classes.
+        x (sparse.csr_matrix): A matrix with dimensions number of instances * number of features.
+        options (str): The option string passed to liblinear.
+
+    Returns:
+        A model which can be used in predict_values.
+    """
+    x, options, bias = prepare_options(x, options)
+    num_labels = y.shape[1]
+    y = np.squeeze(y.nonzero()[1])
+
+    with silent_stderr():
+        model = train(y, x, options)
+
+    train_labels = model.get_labels()
```

Comment:
this and next lines can be combined
train_labels = np.array(...
"ind" not a good name. Let's just use train_labels subsequently